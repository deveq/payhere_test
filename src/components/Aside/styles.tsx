import styled from 'styled-components';
import {flexCenter} from 'styles/helper';

export const Aside = styled.aside`
    width: 200px;
    background: ${({theme}) => theme.bg_page1};
    padding: 30px;
    box-sizing: border-box;
    flex-shrink: 0;
`;

export const List = styled.ul`
    display: grid;
    gap: 5px;
`;

export const Title = styled.h2`
    font-size: 24px;
    font-weight: bold;
    border-bottom: ${({theme}) => `1px solid ${theme.bg_element4}`};
    color: ${({theme}) => theme.text1};
    padding-bottom: 10px;
    margin-bottom: 20px;
    text-indent: 10px;
`;

export const ListItem = styled.li<{selected: boolean}>`
    background: ${({selected,theme}) => selected ? theme.bg_element1 : 'transparent'};
    padding: 15px 20px;
    border-radius: 10px;
    font-weight: bold;
    color: ${({selected, theme}) => selected ? theme.text2 : theme.text3};
    &:hover,
    &:active {
        background: ${({theme}) => theme.bg_element1};
    };
    cursor: pointer;
`;

export const Subtitle = styled.div`

`;