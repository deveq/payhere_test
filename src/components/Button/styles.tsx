import styled from "styled-components";

type ButtonSize = 'sm' | 'md' | 'lg';
export interface ButtonProps {
    size?: ButtonSize;
    backgroundColor?: string;
    borderColor?: string;
    color?: string;
    showShadow?: boolean;
}

const handleSize = (size?: ButtonSize) => {
    if (size === 'sm') return '0.5rem 1rem';
    if (size === 'md') return '0.75rem 1.5rem';
    if (size === 'lg') return '1rem 2rem';
    return '0.75rem 1.5rem';
}

export const Button = styled.button<ButtonProps>`
    background: ${({backgroundColor, theme}) => 
        backgroundColor || theme.bg_element2
    };
    border-radius: 5px;
    color: ${({color, theme}) => 
        color || theme.text1
    };
    padding: ${({size}) => handleSize(size)};
    font-weight: bold;
    box-shadow: ${({theme, showShadow}) => showShadow ? theme.boxShadow : 'none'};
    border: ${({showShadow, theme}) => showShadow ? 'none' : `1px solid ${theme.border2}` };
    &:hover,
    &:active {
        background: ${({theme}) => theme.bg_element3};
    }
    cursor: pointer;
`;

Button.defaultProps = {
    size: 'md',
}

export const Primary = styled(Button)`
    background: ${({theme}) => theme.primary};
    color: white;
`;

export const Destructive = styled(Button)`
    background: ${({theme}) => theme.destructive};
    color: white;
    border: none;
    &:hover,
    &:active {
        background: ${({theme}) => theme.destructive};
    }
`;