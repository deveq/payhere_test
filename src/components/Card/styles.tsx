import styled from 'styled-components';

export const CardContainer = styled.div`
    background: ${({theme}) => theme.bg_element1};
    border-radius: 10px;
    padding: 5px 10px;
`;