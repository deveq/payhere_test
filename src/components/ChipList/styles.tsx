import styled from 'styled-components';

interface ChipProps {
    selected?: boolean;
}

export const Chip = styled.div<ChipProps>`
    padding: 5px 10px;
    max-width: 150px;
    min-width: 40px;
    text-align: center;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
    border-radius: 20px;
    border: 1px solid ${({theme}) => theme.border2};
    background: ${({selected, theme}) => selected ? theme.primary : 'white'};
    color: ${({selected}) => selected ? 'white' : '#333'};
    cursor: pointer;
    &:active {
        opacity: 0.9;
    }
`;

export const ChipList = styled.div`
    display: flex;
    /* flex-direction: column;
    align-items: flex-start; */
    flex-wrap: wrap;
    gap: 8px;
    margin-bottom: 10px;
`;

export const Message = styled.p`
    color: ${({theme}) => theme.text1};
`;