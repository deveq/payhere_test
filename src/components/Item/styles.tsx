import styled from 'styled-components';

export const Container = styled.li`
    display: flex;
    flex-direction: column;
    background: ${({theme}) => theme.bg_page2};
    border: 1px solid ${({theme}) => theme.border3};
    border-radius: 10px;
    padding: 20px 10px;
`;

export const Body = styled.div`
    display: flex;
    flex-direction: column;
    gap: 5px;
`;

export const TopInfo = styled.div`
    color: ${({theme}) => theme.text2};
    font-weight: bold;
    font-size: 14px;
`;

export const BottomInfo = styled.div`
    color: ${({theme}) => theme.text2};
    font-weight: bold;
    font-size: 14px;
`;

export const Title = styled.h4`
    color: ${({theme}) => theme.primary};
    font-weight: bold;
    cursor: pointer;
`;

export const Description = styled.p`
    color: ${({theme}) => theme.text1};
    font-size: 14px;
    max-height: 40px;
    overflow: hidden;
    text-overflow: ellipsis;
    -webkit-line-clamp: 3;
    -webkit-box-orient: vertical;
`;

export const Meta = styled.div`
    color: ${({theme}) => theme.text2};
    font-size: 12px;
`;
