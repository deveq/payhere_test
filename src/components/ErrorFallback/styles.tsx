import styled from 'styled-components';
import {flexCenter} from 'styles/helper';
import {FaRegSadTear} from 'react-icons/fa';

export const Container = styled.div`
    height: 100vh;
    ${flexCenter};
    flex-direction: column;
`;

export const Paragraph = styled.p`
    font-size: 30px;
`;

export const ErrorMessage = styled.pre`
    font-size: 20px;
`;

export const SadIcon = styled(FaRegSadTear)`
    width: 100px;
    height: 100px;
    color: gray;
`;