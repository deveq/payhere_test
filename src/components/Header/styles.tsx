import React from 'react';
import styled from 'styled-components';
import { ReactComponent } from 'assets/logo.svg';
import {flexCenter} from 'styles/helper';


export const Container = styled.header`
	display: flex;
	padding: 1rem 2rem;
	background: ${(props) => props.theme.bg_page1};
	box-shadow: ${(props) => props.theme.boxShadow};
`;

export const Logo = styled(ReactComponent)`
	width: 50px;
	height: 50px;
	${flexCenter};
`;

export const Headline = styled.h1`
`;

export const Nav = styled.nav`
	flex: 1;
	${flexCenter};
`;

export const LoginContainer = styled.div`

`;