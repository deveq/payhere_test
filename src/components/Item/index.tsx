import * as Styled from './styles';

interface ItemProps {
	title: string;
	description?: string | null;
	meta: string;
	upperInfo?: string;
	lowerInfo?: string;
	onClick: () => void;
}

const Item = ({ upperInfo, meta, title, lowerInfo, description, onClick }: ItemProps) => {
	return (
		<Styled.Container>
			<Styled.TopInfo>{upperInfo}</Styled.TopInfo>
			<Styled.Body>
				<Styled.Title onClick={onClick}>{title}</Styled.Title>
				<Styled.Description>{description}</Styled.Description>
			</Styled.Body>
			<Styled.BottomInfo>{lowerInfo}</Styled.BottomInfo>
			<Styled.Meta>{meta}</Styled.Meta>
		</Styled.Container>
	);
};

export default Item;
