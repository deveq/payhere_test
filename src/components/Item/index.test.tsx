import { screen, fireEvent } from '@testing-library/react';
import { renderWithWrapper } from 'tests';
import Item from './index';
import { lightTheme } from 'styles/theme';
import 'jest-styled-components';

describe('<Item />', () => {
	const titleText = 'title';
	const metaText = 'meta';
	const descriptionText = 'description';
	const lowerInfoText = 'lower';
	const upperInfoText = 'upper';

	it('정상적으로 렌더링되는지 확인한다', () => {
		const onClick = () => {};
		const { container } = renderWithWrapper(
			<Item onClick={onClick} title={titleText} meta={metaText} />,
		);

		const item = screen.getByRole('listitem');
		expect(item).toBeInTheDocument();
		expect(item).toHaveStyle({
			background: lightTheme.bg_page2,
			border: `1px solid ${lightTheme.border3}`,
		});
		expect(container).toMatchSnapshot();
	});

	it('props에 따라 정상적으로 렌더링되는지 확인한다', () => {
		const onClick = () => {};
		renderWithWrapper(
			<Item
				onClick={onClick}
				title={titleText}
				meta={metaText}
				description={descriptionText}
				lowerInfo={lowerInfoText}
				upperInfo={upperInfoText}
			/>,
		);

		const title = screen.getByText(titleText);
		const meta = screen.getByText(metaText);
		const description = screen.getByText(descriptionText);
		const lowerInfo = screen.getByText(lowerInfoText);
		const upperInfo = screen.getByText(upperInfoText);

		expect(title).toBeInTheDocument();
		expect(meta).toBeInTheDocument();
		expect(description).toBeInTheDocument();
		expect(lowerInfo).toBeInTheDocument();
		expect(upperInfo).toBeInTheDocument();

		expect(meta).toHaveStyleRule('color', lightTheme.text2);

		expect(lowerInfo).toHaveStyleRule('color', lightTheme.text2);

		expect(title).toHaveStyleRule('color', lightTheme.primary);

		expect(description).toHaveStyleRule('color', lightTheme.text1);
	});

	it('click 이벤트가 정상동작하는지 확인한다', () => {
		const fn = jest.fn();
		renderWithWrapper(<Item title={titleText} meta={metaText} onClick={fn} />);
		expect(fn).toHaveBeenCalledTimes(0);
		const title = screen.getByText(titleText);
		fireEvent.click(title);
		expect(fn).toHaveBeenCalledTimes(1);
	});
});
