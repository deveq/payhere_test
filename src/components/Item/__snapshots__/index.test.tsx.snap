// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> 정상적으로 렌더링되는지 확인한다 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: #FFFFFF;
  border: 1px solid #DEE2E6;
  border-radius: 10px;
  padding: 20px 10px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 5px;
}

.c1 {
  color: #495057;
  font-weight: bold;
  font-size: 14px;
}

.c5 {
  color: #495057;
  font-weight: bold;
  font-size: 14px;
}

.c3 {
  color: #216fdb;
  font-weight: bold;
  cursor: pointer;
}

.c4 {
  color: #212529;
  font-size: 14px;
  max-height: 40px;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
}

.c6 {
  color: #495057;
  font-size: 12px;
}

<div>
  <li
    class="c0"
  >
    <div
      class="c1"
    />
    <div
      class="c2"
    >
      <h4
        class="c3"
      >
        title
      </h4>
      <p
        class="c4"
      />
    </div>
    <div
      class="c5"
    />
    <div
      class="c6"
    >
      meta
    </div>
  </li>
</div>
`;
